activerecord/lib/active_record/encryption/extended_deterministic_queries.rb

```diff
@@ -26,7 +26,7 @@ def self.install_support
         # some prepared statements caching. That's why we need to intercept +ActiveRecord::Base+ as soon
         # as it's invoked (so that the proper prepared statement is cached).
         ActiveRecord::Relation.prepend(RelationQueries)
-        ActiveRecord::Base.include(CoreQueries)
+        ActiveSupport.on_load(:active_record) { include CoreQueries }
```

Comment:
We should not use `on_load` inside here. Can we move to the caller?

Reply:
Done.